src/report-filter-controls.tsx

```diff
@@ -57,13 +57,26 @@ function ReportFilterControls({
     setAllAgencies,
   };
 
-  function update(event: Event) {
-    const form = event.currentTarget as HTMLFormElement;
+  function update(event: Event, overrideFormData = {}) {
+    const form =
+      (event.target instanceof HTMLButtonElement && event.target.form) ||
+      (event.currentTarget as HTMLFormElement);
     const formData = Array.from(new FormData(form)) as string[][];
-    route(`${path}?${new URLSearchParams(formData).toString().replace(/\+/g, "%20")}`);
+    const seachParams = new URLSearchParams(formData);
+    Object.entries(overrideFormData).forEach(([key, value]) => seachParams.set(key, String(value)));
+    route(`${path}?${seachParams.toString().replace(/\+/g, "%20")}`);
     event.preventDefault();
   }
 
+  function updateTimeRange(interval: CountableTimeInterval, offset: number) {
+    return function (event: Event) {
+      return update(event, {
```

Comment:
Not sure if it's any better / easier, but an alternative is that we could actually change the date input values directly, then rely on the bubbled change event to call `update` indirectly. More I think on it, more I think it might not be worthwhile, since we don't have an easy way to trigger an update to a field value except through `update`.